controllers/vspherevm_controller.go

```diff
@@ -223,6 +224,16 @@ func (r vmReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr error)
 		return r.reconcileDelete(vmContext)
 	}
 
+	cluster, err := clusterutilv1.GetClusterFromMetadata(r.ControllerContext, r.Client, vsphereVM.ObjectMeta)
```

Comment:
Didn't we want to do this before calling either reconcileDelete or reconcileNormal? cc @vincepri

Reply:
We're likely going to have to be careful with that (not just here, but other places as well), otherwise we may end up in a situation where we can't delete a resource if the Cluster resource is not present.

Reply:
if we check for pause before the delete (and optionally go through and continue if we can't find the cluster) we can endup in `reconcileNormal` which needs the `Cluster` object. A solution could be to recheck for `err` before reconcile normal, wdyt @detiber @ncdc